### PR TITLE
Add stm hs

### PR DESCRIPTION
**Pull request by UweBonnes**
**Describe the PR**
Allow a HS port to enumerate as FS port
Add STM32H745I-DISCO

### CODE REVIEW

File: hw/bsp/stlinkv3mini/board.mk [hunk 1]
```diff
@@ -1,3 +1,6 @@
+PORT ?= 1
```
[the remark this answers is not in the document]
Comment (UweBonnes) [reply hw/bsp/stlinkv3mini/board.mk]: I removed the senseless port selection and use port1 fixed mow

File: hw/bsp/stlinkv3mini/board.mk [hunk 2]
```diff
@@ -9,8 +12,19 @@ CFLAGS += \
...
   -DHSE_VALUE=25000000 \
   -DCFG_TUSB_MCU=OPT_MCU_STM32F7 \
-  -DBOARD_DEVICE_RHPORT_NUM=1 \
-  -DBOARD_DEVICE_RHPORT_SPEED=OPT_MODE_HIGH_SPEED
+  -DBOARD_DEVICE_RHPORT_NUM=$(PORT)
+
+ifeq ($(PORT), 1)
+ ifeq ($(REDUCE_SPEED), 0)
+CFLAGS += -DBOARD_DEVICE_RHPORT_SPEED=OPT_MODE_HIGH_SPEED
+$(info "OTG_HS/HS")
```
[the remark this answers is not in the document]
Comment (UweBonnes) [reply hw/bsp/stlinkv3mini/board.mk]: I use "Using OTG_HS in HS mode" and "Using OTG_HS in FS mode" now

File: hw/bsp/stm32f723disco/board.mk [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# Default is Highspeed port
-PORT ?= 1
+PORT ?= 0
```
[the remark this answers is not in the document]
Comment (UweBonnes) [reply hw/bsp/stm32f723disco/board.mk]: Default is now OTG_HS

**Comment by hathach** [comment]
@UweBonnes since you are actively working on this, I convert this PR to draft for now, feel free to make it ready when you think it is ready for the merge. Hope you don't mind this.

**Comment by UweBonnes** [comment]
I think it is ready, when you think your remarks have be implemented acceptable. I only addes another BSP, this tome for the stm32l4r5nucleo.

File: hw/bsp/stlinkv3mini/board.mk [hunk 4]
```diff
@@ -1,3 +1,7 @@
+# Only OTG-HS has a connector on this board
+
+REDUCE_SPEED ?= 0
```
Comment (hathach) [inline hw/bsp/stlinkv3mini/board.mk:3]: could you change this option to `SPEED=full` or `SPEED=high` to make it clearer for user.

**Pull request merged**